src/arcsf/config/experiment.py

```diff
@@ -0,0 +1,304 @@
+            f"{_make_config_name(top_config_name, n)}.yaml"
+        )
+        with open(file_name, "w") as f:
+            yaml.dump(combo, f)
+
+    # Check whether to generate baskerville scripts:
+    if top_config["use_bask"]:
+
+        # Create directory for train script for experiment if it doesn't exist
+        train_dir = os.path.join(arcsf.constants.PROJECT_DIR, "train_scripts")
+        if not os.path.exists(train_dir):
+            os.mkdir(train_dir)
+
+        # Get jinja template
+        environment = Environment(
+            loader=FileSystemLoader(
+                os.path.join(arcsf.constants.PROJECT_DIR, "src", "arcsf", "config")
+            )
+        )
+        template = environment.get_template("jobscript_template.sh")
+
+        # Generate and save train script
+        job_name = f"{top_config_name}_train"
+        train_script = template.render(
+            job_name=job_name,
+            walltime=top_config["bask"]["walltime"],
+            node_number=top_config["bask"]["node_number"],
+            gpu_number=top_config["bask"]["gpu_number"],
+            cpu_per_gpu=top_config["bask"]["cpu_per_gpu"],
+            array_number=len(all_combinations) - 1,
+            script_name="scripts/train.py",
+            experiment_file=f"{top_config_name}/experiment",
+        )
+        with open(os.path.join(train_dir, f"{job_name}.sh"), "w") as f:
+            f.write(train_script)
+
+
+class ExperimentConfig(Config):
+    """Experiment config class.
+
+    Attributes:
+        data_config: Dataset config class.
+        model_config: Model config class, covering both model and hyperparameters.
+        train_type: Either "retain" or "full": whether fine-tuning on a full dataset
+                    or a retain set.
+        full_model_name: If train_type is "retain", the name of the experiment yaml
+                         file used for fine-tuning the full model to be compared against
+        use_wandb: Whether or not to use wandb for logging
+        wandb_config: Dict of wandb arguments
+        run_name: Name of the run logged to wandb
+        seed: Seed for random generation
+    """
+
+    def __init__(
+        self,
+        data_config: str,
+        model_config: str,
+        hyperparameter_config: str,
+        train_type: str,
+        use_wandb: bool,
+        wandb_config: dict | None,
+        seed: int | None,
+        full_model_name: str | None = None,
+    ) -> None:
+        super().__init__()
+
+        # Load in other configsg
+        self.data_config = DataConfig.from_yaml(
+            os.path.join(arcsf.constants.DATA_CONFIG_DIR, f"{data_config}.yaml")
+        )
+        model_dir = os.path.join(arcsf.constants.MODEL_CONFIG_DIR, model_config)
+        self.model_config = ModelConfig.from_yaml(
+            os.path.join(model_dir, f"{model_config}.yaml"),
+            os.path.join(model_dir, "hyperparameters", f"{hyperparameter_config}.yaml"),
+        )
+
```

Comment:
Since the constants are `Path`s you should be able to do this (just a bit neater):


```suggestion
        # Load in other configs
        self.data_config = DataConfig.from_yaml(
            arcsf.constants.DATA_CONFIG_DIR / f"{data_config}.yaml"
        )
        model_dir = arcsf.constants.MODEL_CONFIG_DIR / model_config
        self.model_config = ModelConfig.from_yaml(
            model_dir / f"{model_config}.yaml",
            model_dir / "hyperparameters" / f"{hyperparameter_config}.yaml",
        )
```